### PR TITLE
chore(auth-server): update stripe

### LINKED ISSUE #7436: chore(deps): bump stripe
from #7433

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-2985)
┆Issue Number: FXA-2985

### PR DESCRIPTION
**Pull request by bbangert**
Because:

* We want to use the latest library and API to avoid falling behind.

This commit:

* Updates to the latest Stripe library.

Closes #7436

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

**Comment by dannycoates** [comment]
I think with a rebase this'll pass ci

File: packages/fxa-auth-server/lib/payments/stripe.ts [hunk 1]
```diff
@@ -1765,19 +1764,20 @@ export class StripeHelper {
       metadata: { userid: uid },
     } = customer;
 
-    if (!subscription.plan) {
+    const subItems = subscription.items.data;
+    if (subItems.length > 1) {
```
Comment (chenba) [inline packages/fxa-auth-server/lib/payments/stripe.ts:1768]: Use `singlePlan` here?

**Review by chenba — APPROVED** [review]
LGTM

**Comment by bbangert** [comment]
@dannycoates nope, tried a rebase. This one has been tricky to resolve locally as well, I'm having difficulty running just this one test with the debugger going to track down why its getting an internal server error.

**Pull request merged**